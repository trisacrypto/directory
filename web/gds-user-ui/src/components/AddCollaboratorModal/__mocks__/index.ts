
export const collaboratorMockValue: any = {
    data: [
        {
            id: "1",
            name: "test",
            email: "test@local.dev",
            roles: ["admin"],
            created_at: "2021-01-01",
        },
    ],
};
