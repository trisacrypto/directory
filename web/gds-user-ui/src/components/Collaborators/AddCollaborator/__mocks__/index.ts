
export const collaboratorMockValue: any = {
    data: [
        {
            id: "1",
            name: "test",
            email: "test@local.dev",
            roles: ["admin"],
            created_at: "2021-01-01",
        },
    ],
};

export const userOrganizationMockValue: any = {
    data: {
        id: "1",
        name: "test",
        domain: "test.local.dev",
        created_at: "2021-01-01",
    },
};
