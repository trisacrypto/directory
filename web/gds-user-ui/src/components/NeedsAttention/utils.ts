export const enum AttentionAction {
  'VERIFY_EMAILS' = 'VERIFY_EMAILS',
  'VERIFY_PHONES' = 'VERIFY_PHONES',
  'VERIFY_ADDRESSES' = 'VERIFY_ADDRESSES',
  'VERIFY_IDENTITIES' = 'VERIFY_IDENTITIES',
  'VERIFY_DOCUMENTS' = 'VERIFY_DOCUMENTS',
  'VERIFY_ALL' = 'VERIFY_ALL',
  'VERIFY_NONE' = 'VERIFY_NONE'
}
