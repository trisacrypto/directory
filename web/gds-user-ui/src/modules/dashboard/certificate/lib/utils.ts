import { basicDetailsValidationSchema } from 'modules/dashboard/certificate/lib/basicDetailsValidationSchema';
import { legalPersonValidationSchemam } from 'modules/dashboard/certificate/lib/legalPersonValidationSchema';
import { contactsValidationSchema } from 'modules/dashboard/certificate/lib/contactsValidationSchema';
import { trisaImplementationValidationSchema } from 'modules/dashboard/certificate/lib/trisaImplementationValidationSchema';
import { trixoQuestionnaireValidationSchema } from 'modules/dashboard/certificate/lib/trixoQuestionnaireValidationSchema';

export const isBasicDetailsCompleted = async (data: any) => {
  try {
    const r = await basicDetailsValidationSchema.validate(data);
    console.log(r);
    return true;
  } catch (err) {
    return false;
  }
};

export const isLegalPersonCompleted = async (data: any) => {
  try {
    const r = await legalPersonValidationSchemam.validate(data);
    console.log(r);
    return true;
  } catch (err) {
    console.log('legal person validation failed', err);
    return false;
  }
};

export const isContactsCompleted = async (data: any) => {
  try {
    const r = await contactsValidationSchema.validate(data);
    console.log(r);
    return true;
  } catch (err) {
    console.log('contacts validation failed', err);
    return false;
  }
};

export const isTrisaImplementationCompleted = async (data: any) => {
  try {
    await trisaImplementationValidationSchema.validate(data);
    return true;
  } catch (err) {
    return false;
  }
};

export const isTrixoQuestionnaireCompleted = async (data: any) => {
  try {
    await trixoQuestionnaireValidationSchema.validate(data);
    return true;
  } catch (err) {
    return false;
  }
};
