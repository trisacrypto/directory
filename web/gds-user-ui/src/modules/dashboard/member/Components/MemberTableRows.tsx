import { Td, Tr, Button, HStack, chakra } from '@chakra-ui/react';

import UnverifiedMember from '../UnverifiedMember';
import { BsEye } from 'react-icons/bs';

interface MemberTableRowsProps {
  rows: any;
}

const MemberTableRows: React.FC<MemberTableRowsProps> = (rows: any) => {
  // for now we are just displaying the unverified member component
  // with the status check story we will check if the member is verified or not and display the appropriate component
  return (
    <Tr>
      {rows.length > 0 ? (
        <>
          <Td>
            <chakra.span display="block"></chakra.span>
            <chakra.span display="block" fontSize="sm" color="gray.700"></chakra.span>
          </Td>
          <Td></Td>
          <Td></Td>
          <Td></Td>
          <Td></Td>
          <Td paddingY={0}>
            <HStack width="100%" justifyContent="center" alignItems="center">
              <Button
                color="blue"
                as={'a'}
                href={``}
                bg={'transparent'}
                _hover={{
                  bg: 'transparent'
                }}
                _focus={{
                  bg: 'transparent'
                }}>
                <BsEye fontSize="24px" />
              </Button>
            </HStack>
          </Td>
        </>
      ) : (
        <Td colSpan={6}>
          <UnverifiedMember />
        </Td>
      )}
    </Tr>
  );
};

export { MemberTableRows };
