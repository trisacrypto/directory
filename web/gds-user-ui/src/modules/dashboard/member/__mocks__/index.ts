export const mainnetMembersMockValue: any = {
  vasps: [
    {
      id: 'fe29b582-20b3-4b75-acb2-4d014af25f28',
      registered_directory: 'vaspdirectory.net',
      common_name: 'mainnet.firecoinex.co',
      endpoint: 'mainnet.firecoinex.co:443',
      name: 'FireCoin Exchange',
      website: 'https://www.firecoinex.co/',
      country: 'SG',
      business_category: 'BUSINESS_ENTITY',
      vasp_categories: ['OTC'],
      verified_on: '2022-04-20T10:45:33Z',
      status: 'VERIFIED',
      first_listed: '2022-04-20T11:10:06Z',
      last_updated: '2022-11-19T09:32:45Z'
    },
    {
      id: '52842ac3-7d9d-4520-8bfb-e6083dbdc8aa',
      registered_directory: 'vaspdirectory.net',
      common_name: 'mainnet.testmachine.com',
      endpoint: 'mainnet.testmachine.com:443',
      name: 'Test Machine',
      website: 'https://testmachine.com',
      country: 'US',
      business_category: 'BUSINESS_ENTITY',
      vasp_categories: ['Exchange', 'Other'],
      verified_on: '2023-03-22T03:55:00Z',
      status: 'VERIFIED',
      first_listed: '2023-03-22T04:22:18Z',
      last_updated: '2023-05-10T04:44:00Z'
    },
    {
      id: 'ec6f2056-726b-4e8c-a916-f7359f6f5581',
      registered_directory: 'vaspdirectory.net',
      common_name: 'mainnet.newcoinex.ai',
      endpoint: 'mainnet.newcoinex.ai:8221',
      name: 'New Coin Exchange',
      website: 'https://newcoinex.ai/',
      country: 'US',
      business_category: 'BUSINESS_ENTITY',
      vasp_categories: ['Other'],
      verified_on: '2023-01-20T12:55:03Z',
      status: 'VERIFIED',
      first_listed: '2023-01-20T14:10:26Z',
      last_updated: '2023-06-18T10:30:01Z'
    }
  ],
  next_page_token: ''
};

export const testnetMembersMockValue: any = {
  vasps: [
    {
      id: '1b99f17d-3441-4885-b9df-8f3475f7e1b4',
      registered_directory: 'trisatest.net',
      common_name: 'trisa-travelrule.sendcoin.io',
      endpoint: 'trisa-travelrule.sendcoin.io:443',
      name: 'SendCoin VASP',
      website: 'https://www.sendcoin.io/',
      country: 'SG',
      business_category: 'BUSINESS_ENTITY',
      vasp_categories: ['OTC'],
      verified_on: '2022-02-10T17:12:23Z',
      status: 'VERIFIED',
      first_listed: '2022-02-10T13:43:55Z',
      last_updated: '2023-04-09T04:02:50Z'
    },
    {
      id: 'aa4c6714-49dc-411c-adfa-b8edc4d58cd7',
      registered_directory: 'trisatest.net',
      common_name: 'testing.example.com',
      endpoint: 'testing.example.com:443',
      name: 'Example Crypto',
      website: 'https://example.com',
      country: 'US',
      business_category: 'BUSINESS_ENTITY',
      vasp_categories: ['Exchange', 'Other'],
      verified_on: '2021-12-07T20:22:00Z',
      status: 'VERIFIED',
      first_listed: '2021-12-01T23:22:18Z',
      last_updated: '2023-01-23T19:19:43Z'
    },
    {
      id: '688059d6-9d14-4b49-8435-f641ba1dec3a',
      registered_directory: 'trisatest.net',
      common_name: 'testnet.spudcoin.ai',
      endpoint: 'testnet.spudcoin.ai:8221',
      name: 'SpudCoin',
      website: 'https://spudcoin.ai/',
      country: 'US',
      business_category: 'BUSINESS_ENTITY',
      vasp_categories: ['Other'],
      verified_on: '2021-07-29T19:11:03Z',
      status: 'VERIFIED',
      first_listed: '2021-07-23T17:10:26Z',
      last_updated: '2022-12-27T18:59:01Z'
    },
    {
      id: '62291255-1ea2-4932-8248-22af4abde298',
      registered_directory: 'trisatest.net',
      common_name: 'test.signal.co.fr',
      endpoint: 'test.signal.co.fr:9212',
      name: 'Signal Coin France',
      website: 'https://ciphertrace.com/',
      country: 'US',
      business_category: 'BUSINESS_ENTITY',
      vasp_categories: ['Other'],
      verified_on: '2021-06-23T17:46:10Z',
      status: 'VERIFIED',
      first_listed: '2021-06-23T17:12:00Z',
      last_updated: '2022-12-27T18:59:22Z'
    },
    {
      id: 'dae1555d-e4cf-4bfb-9858-9b86db71ccb6',
      registered_directory: 'trisatest.net',
      common_name: 'testnet.bitfriend.link',
      endpoint: 'testnet.bitfriend.link:443',
      name: 'BitFriendly',
      website: 'https://bitfriend.link/',
      country: 'US',
      business_category: 'BUSINESS_ENTITY',
      vasp_categories: ['Kiosk'],
      verified_on: '2021-09-14T10:40:30Z',
      status: 'VERIFIED',
      first_listed: '2021-09-01T19:46:04Z',
      last_updated: '2023-01-23T19:29:43Z'
    }
  ],
  next_page_token: 'mLB9CU8O8xQj2XEyjAtlfvTj9imoXnLv/1p8fTLchTg='
};

export const getMockValue = (network: string): any => {
  switch (network) {
    case 'mainnet':
      return mainnetMembersMockValue;
    case 'testnet':
      return testnetMembersMockValue;
    default:
      return mainnetMembersMockValue;
  }
};
