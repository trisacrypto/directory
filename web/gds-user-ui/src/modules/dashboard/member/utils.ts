import { DirectoryType, VaspDirectoryEnum, DirectoryTypeEnum } from './memberType';
import { convertToCVS, downloadCSV } from 'utils/utils';
import { t } from '@lingui/macro';

export const memberTableHeader = [
  {
    key: 'name',
    label: t`Member Name`
  },
  {
    key: 'first_listed',
    label: t`Joined`
  },
  {
    key: 'last_updated',
    label: t`Last Updated`
  },
  {
    key: 'registered_directory',
    label: t`Network`
  },
  {
    key: 'status',
    label: t`Status`
  }
];
export const getVaspDirectory = (dir: DirectoryType) => {
  return dir === DirectoryTypeEnum.TESTNET ? VaspDirectoryEnum.TESTNET : VaspDirectoryEnum.MAINNET;
};

export const getVaspNetwork = (dir: any) => {
  switch (dir) {
    case (VaspDirectoryEnum.TESTNET, VaspDirectoryEnum.TESTNET_DEV):
      return 'TestNet';
    case (VaspDirectoryEnum.MAINNET, VaspDirectoryEnum.MAINNET_DEV):
      return 'MainNet';
    default:
      return 'MainNet';
  }
};

export const getVaspStatus = (status: number) => {
  switch (status) {
    case 1:
      return t`NO VERIFICATION`;
    case 2:
      return t`SUBMITTED`;
    case 3:
      return t`EMAIL VERIFIED`;
    case 4:
      return t`PENDING REVIEW`;
    case 5:
      return t`REVIEWED`;
    case 6:
      return t`VERIFIED`;
    case 7:
      return t`REJECTED`;
    case 8:
      return t`APPEALED`;
    case 9:
      return t`ERRORED`;
    default:
      return t`NO VERIFICATION`;
  }
};

export const downloadMembers2CVS = (member: any) => {
  const m = member.map((item: any) => {
    return {
      ...item,
      status: getVaspStatus(item.status),
      registered_directory: getVaspNetwork(item.registered_directory)
    };
  });
  const memberCsv = convertToCVS(m, memberTableHeader as ITableHeader[]);
  downloadCSV(memberCsv, 'members');
};

