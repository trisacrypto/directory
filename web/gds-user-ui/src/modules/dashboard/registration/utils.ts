import { getDefaultValue } from 'components/BasicDetailsForm/validation';
import { getRegistrationDefaultValues } from 'modules/dashboard/certificate/lib';
import {
  postRegistrationData,
  getRegistrationData,
  getSubmissionStatus
} from 'modules/dashboard/registration/service';
import { handleError } from 'utils/utils';

export const postRegistrationValue = (data: any) => {
  console.log('[postRegistrationValue]', data);
  return new Promise((resolve, reject) => {
    postRegistrationData(data)
      .then((res) => {
        console.log('[postRegistrationData]', res);
        if (res.status === 204) {
          resolve(res);
        } else {
          reject(res);
        }
      })
      .catch((err) => {
        console.log('[postRegistrationData]', err);
        reject(err);
      });
  });
};
export const getRegistrationDefaultValue = async () => {
  try {
    const regData = await getRegistrationData();
    if (regData.status === 200 && Object.keys(regData.data).length > 0) {
      return regData.data;
    }
    const defaultValue: any = localStorage.getItem('certificateForm');
    if (defaultValue) {
      const val = JSON.parse(defaultValue);
      const postData = await postRegistrationData(val);
      if (postData.status === 204) {
        const getData = await getRegistrationData();
        localStorage.removeItem('certificateForm');
        return getData.data;
      }
    }
    const v = getRegistrationDefaultValues();
    await postRegistrationValue(v);
    return getRegistrationDefaultValues();
  } catch (err: any) {
    handleError(err, 'failed to get registration data');
  }
};

export const setRegistrationDefaultValue = () => {
  console.log('[setRegistrationDefaultValue]');
  const defaultValue: any = getRegistrationDefaultValues();
  return new Promise((resolve, reject) => {
    postRegistrationData(defaultValue)
      .then((res) => {
        // console.log('[default postRegistration value]', res);
        if (res.status === 204) {
          resolve(res);
        } else {
          reject(res);
        }
      })
      .catch((err) => {
        handleError(err, 'failed to post registration value');
        reject(err);
      });
  });
};

// get registration data from backend and download the file
export const downloadRegistrationData = async () => {
  try {
    const regData = await getRegistrationData();
    if (regData.status === 200 && Object.keys(regData.data).length > 0) {
      const blob = new Blob([JSON.stringify(regData.data)], { type: 'application/json' });
      const url = window.URL.createObjectURL(blob);
      const link = document.createElement('a');
      link.href = url;
      link.setAttribute('download', 'registration.json');
      document.body.appendChild(link);
      link.click();
    }
  } catch (err: any) {
    handleError(err, 'failed to get registration data');
  }
};

// load default stepper

export const getDefaultStepper = async () => {
  try {
    const [regData, regStatus] = await Promise.all([getRegistrationData(), getSubmissionStatus()]);
    console.log('[regStatus]', regStatus.data);
    if (regData.status === 200 && Object.keys(regData.data).length > 0) {
      return {
        currentStep: regData.data.state.current,
        steps: regData.data.state.steps,
        lastStep: null,
        hasReachSubmitStep: regData.data.state.ready_to_submit,
        testnetSubmitted: regStatus?.data?.testnetSubmitted || false,
        mainnetSubmitted: regStatus?.data?.mainnetSubmitted || false
      };
    }
    const defaultValue: any = {
      current: 1,
      steps: [
        {
          key: 1,
          status: 'progress'
        }
      ]
    };
    // update registrations state object
    const postData = await postRegistrationData({ state: defaultValue });
    if (postData.status === 204) {
      const getData = await getRegistrationData();
      return {
        currentStep: getData.data.state.current,
        steps: getData.data.state.steps,
        lastStep: null,
        hasReachSubmitStep: false,
        testnetSubmitted: false,
        mainnetSubmitted: false
      };
    }
  } catch (err: any) {
    handleError(err, 'failed to get stepper data');
  }
};

// // load default stepper without async call
// export const loadDefaultStepperSync = () => {
//   return Promise.resolve({
//     currentStep: 1,
//     steps: [
//       {
//         key: 1,
//         status: 'progress'
//       }
//     ],
//     lastStep: null,
//     hasReachSubmitStep: false
//   });
// };

// set stepper data
export const getRegistrationAndStepperData = async () => {
  try {
    const [regData, regStatus] = await Promise.all([
      getRegistrationDefaultValue(),
      getSubmissionStatus()
    ]);
    if (regData) {
      const response: any = {
        registrationData: regData,
        stepperData: {
          currentStep: regData?.state?.current || 1,
          steps: regData?.state?.steps || [{ key: 1, status: 'progress' }],
          lastStep: null,
          hasReachSubmitStep: regData?.state?.ready_to_submit || false,
          testnetSubmitted: !!regStatus?.data?.testnet_submitted,
          mainnetSubmitted: !!regStatus?.data?.mainnet_submitted
        }
      };
      return response;
    }
  } catch (err: any) {
    handleError(err, 'failed to get stepper data');
  }
};
