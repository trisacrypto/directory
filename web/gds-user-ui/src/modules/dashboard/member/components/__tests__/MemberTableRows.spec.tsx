import { act, render, screen } from 'utils/test-utils';
import MemberTableRows, { MemberTableRowsProps } from '../MemberTableRows';

import { dynamicActivate } from 'utils/i18nLoaderHelper';
import { mainnetMembersMockValue } from '../../__mocks__';
import { Table, Tbody } from '@chakra-ui/react';

function renderComponent(rowProps?: MemberTableRowsProps) {
  const defaultProps: MemberTableRowsProps = {
    rows: mainnetMembersMockValue.vasps,
    hasError: false,
    isLoading: false
  };
  const props = rowProps || defaultProps;
  return render(
    <Table>
      <Tbody>
        <MemberTableRows {...props} />
      </Tbody>
    </Table>
  );
}

describe('MemberTable', () => {
  beforeAll(() => {
    act(() => {
      dynamicActivate('en');
    });
  });

  it('should render', () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });

  it('should render the mainnet data in the table rows', () => {
    renderComponent();
    const memberName1 = screen.getByText('FireCoin Exchange');
    const joined1 = screen.getByText('Apr 20, 2022');
    const lastUpdated1 = screen.getByText('Nov 19, 2022');
    const memberName2 = screen.getByText('Test Machine');
    const joined2 = screen.getByText('Mar 22, 2023');
    const lastUpdated2 = screen.getByText('May 10, 2023');
    const memberName3 = screen.getByText('New Coin Exchange');
    const joined3 = screen.getByText('Jan 20, 2023');
    const lastUpdated3 = screen.getByText('Jun 18, 2023');
    expect(memberName1).toBeInTheDocument();
    expect(joined1).toBeInTheDocument();
    expect(lastUpdated1).toBeInTheDocument();
    expect(memberName2).toBeInTheDocument();
    expect(joined2).toBeInTheDocument();
    expect(lastUpdated2).toBeInTheDocument();
    expect(memberName3).toBeInTheDocument();
    expect(joined3).toBeInTheDocument();
    expect(lastUpdated3).toBeInTheDocument();
  });

  it('should render the unverified member error', () => {
    const propsMock = {
      rows: [],
      hasError: true,
      isLoading: false
    };
    renderComponent(propsMock);
    const unverifiedMember = screen.getByTestId('unverified-member');
    expect(unverifiedMember).toBeInTheDocument();
  });
});
