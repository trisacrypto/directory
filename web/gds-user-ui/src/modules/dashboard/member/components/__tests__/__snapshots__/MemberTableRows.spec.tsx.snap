// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Member Table Rows should render 1`] = `
.emotion-0 {
  font-variant-numeric: lining-nums tabular-nums;
  border-collapse: collapse;
  width: var(--chakra-sizes-full);
}

.emotion-3 {
  text-align: start;
  -webkit-padding-start: var(--chakra-space-6);
  padding-inline-start: var(--chakra-space-6);
  -webkit-padding-end: var(--chakra-space-6);
  padding-inline-end: var(--chakra-space-6);
  padding-top: var(--chakra-space-4);
  padding-bottom: var(--chakra-space-4);
  line-height: var(--chakra-lineHeights-5);
  border-bottom: var(--chakra-borders-1px);
  border-color: var(--chakra-colors-gray-100);
}

.emotion-3[data-is-numeric=true] {
  text-align: end;
}

.emotion-4 {
  display: block;
}

.emotion-9 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  vertical-align: top;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  max-width: 100%;
  font-weight: var(--chakra-fontWeights-medium);
  line-height: 1.2;
  outline: 2px solid transparent;
  outline-offset: 2px;
  min-height: 1.5rem;
  min-width: 1.5rem;
  font-size: var(--chakra-fontSizes-sm);
  border-radius: var(--chakra-radii-md);
  -webkit-padding-start: var(--chakra-space-2);
  padding-inline-start: var(--chakra-space-2);
  -webkit-padding-end: var(--chakra-space-2);
  padding-inline-end: var(--chakra-space-2);
  background: var(--chakra-colors-green-400);
  color: var(--chakra-colors-white);
}

.emotion-9:focus,
.emotion-9[data-focus] {
  box-shadow: var(--chakra-shadows-outline);
}

.emotion-10 {
  text-align: start;
  -webkit-padding-start: var(--chakra-space-6);
  padding-inline-start: var(--chakra-space-6);
  -webkit-padding-end: var(--chakra-space-6);
  padding-inline-end: var(--chakra-space-6);
  padding-top: 0px;
  padding-bottom: 0px;
  line-height: var(--chakra-lineHeights-5);
  border-bottom: var(--chakra-borders-1px);
  border-color: var(--chakra-colors-gray-100);
}

.emotion-10[data-is-numeric=true] {
  text-align: end;
}

.emotion-11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
}

.emotion-11>*:not(style)~*:not(style) {
  margin-top: 0px;
  -webkit-margin-end: 0px;
  margin-inline-end: 0px;
  margin-bottom: 0px;
  -webkit-margin-start: 0.5rem;
  margin-inline-start: 0.5rem;
}

.emotion-12 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: relative;
  white-space: nowrap;
  vertical-align: middle;
  outline: 2px solid transparent;
  outline-offset: 2px;
  width: auto;
  line-height: 1.2;
  border-radius: var(--chakra-radii-md);
  font-weight: var(--chakra-fontWeights-semibold);
  transition-property: var(--chakra-transition-property-common);
  transition-duration: var(--chakra-transition-duration-normal);
  height: var(--chakra-sizes-10);
  min-width: var(--chakra-sizes-10);
  font-size: var(--chakra-fontSizes-md);
  -webkit-padding-start: var(--chakra-space-4);
  padding-inline-start: var(--chakra-space-4);
  -webkit-padding-end: var(--chakra-space-4);
  padding-inline-end: var(--chakra-space-4);
  background: var(--chakra-colors-transparent);
  color: blue;
}

.emotion-12:focus,
.emotion-12[data-focus] {
  background: var(--chakra-colors-transparent);
}

.emotion-12[disabled],
.emotion-12[aria-disabled=true],
.emotion-12[data-disabled] {
  opacity: 0.4;
  cursor: not-allowed;
  box-shadow: var(--chakra-shadows-none);
}

.emotion-12:hover,
.emotion-12[data-hover] {
  background: var(--chakra-colors-transparent);
}

.emotion-12:active,
.emotion-12[data-active] {
  background: var(--chakra-colors-gray-300);
}

<div>
  <table
    class="chakra-table emotion-0"
    role="table"
  >
    <tbody
      class="emotion-1"
    >
      <tr
        class="emotion-1"
        role="row"
      >
        <td
          class="emotion-3"
          role="gridcell"
        >
          <span
            class="emotion-4"
          >
            FireCoin Exchange
          </span>
        </td>
        <td
          class="emotion-3"
          role="gridcell"
        >
          Apr 20, 2022
        </td>
        <td
          class="emotion-3"
          role="gridcell"
        >
          Nov 19, 2022
        </td>
        <td
          class="emotion-3"
          role="gridcell"
        >
          MainNet
        </td>
        <td
          class="emotion-3"
          role="gridcell"
        >
          <span
            class="emotion-9"
          >
            VERIFIED
          </span>
        </td>
        <td
          class="emotion-10"
          role="gridcell"
        >
          <div
            class="chakra-stack emotion-11"
          >
            <button
              class="chakra-button emotion-12"
              data-testid="member-modal-button"
              type="button"
            >
              <svg
                fill="currentColor"
                font-size="24px"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M16 8s-3-5.5-8-5.5S0 8 0 8s3 5.5 8 5.5S16 8 16 8zM1.173 8a13.133 13.133 0 0 1 1.66-2.043C4.12 4.668 5.88 3.5 8 3.5c2.12 0 3.879 1.168 5.168 2.457A13.133 13.133 0 0 1 14.828 8c-.058.087-.122.183-.195.288-.335.48-.83 1.12-1.465 1.755C11.879 11.332 10.119 12.5 8 12.5c-2.12 0-3.879-1.168-5.168-2.457A13.134 13.134 0 0 1 1.172 8z"
                />
                <path
                  d="M8 5.5a2.5 2.5 0 1 0 0 5 2.5 2.5 0 0 0 0-5zM4.5 8a3.5 3.5 0 1 1 7 0 3.5 3.5 0 0 1-7 0z"
                />
              </svg>
            </button>
          </div>
        </td>
      </tr>
      <tr
        class="emotion-1"
        role="row"
      >
        <td
          class="emotion-3"
          role="gridcell"
        >
          <span
            class="emotion-4"
          >
            Test Machine
          </span>
        </td>
        <td
          class="emotion-3"
          role="gridcell"
        >
          Mar 22, 2023
        </td>
        <td
          class="emotion-3"
          role="gridcell"
        >
          May 10, 2023
        </td>
        <td
          class="emotion-3"
          role="gridcell"
        >
          MainNet
        </td>
        <td
          class="emotion-3"
          role="gridcell"
        >
          <span
            class="emotion-9"
          >
            VERIFIED
          </span>
        </td>
        <td
          class="emotion-10"
          role="gridcell"
        >
          <div
            class="chakra-stack emotion-11"
          >
            <button
              class="chakra-button emotion-12"
              data-testid="member-modal-button"
              type="button"
            >
              <svg
                fill="currentColor"
                font-size="24px"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M16 8s-3-5.5-8-5.5S0 8 0 8s3 5.5 8 5.5S16 8 16 8zM1.173 8a13.133 13.133 0 0 1 1.66-2.043C4.12 4.668 5.88 3.5 8 3.5c2.12 0 3.879 1.168 5.168 2.457A13.133 13.133 0 0 1 14.828 8c-.058.087-.122.183-.195.288-.335.48-.83 1.12-1.465 1.755C11.879 11.332 10.119 12.5 8 12.5c-2.12 0-3.879-1.168-5.168-2.457A13.134 13.134 0 0 1 1.172 8z"
                />
                <path
                  d="M8 5.5a2.5 2.5 0 1 0 0 5 2.5 2.5 0 0 0 0-5zM4.5 8a3.5 3.5 0 1 1 7 0 3.5 3.5 0 0 1-7 0z"
                />
              </svg>
            </button>
          </div>
        </td>
      </tr>
      <tr
        class="emotion-1"
        role="row"
      >
        <td
          class="emotion-3"
          role="gridcell"
        >
          <span
            class="emotion-4"
          >
            New Coin Exchange
          </span>
        </td>
        <td
          class="emotion-3"
          role="gridcell"
        >
          Jan 20, 2023
        </td>
        <td
          class="emotion-3"
          role="gridcell"
        >
          Jun 18, 2023
        </td>
        <td
          class="emotion-3"
          role="gridcell"
        >
          MainNet
        </td>
        <td
          class="emotion-3"
          role="gridcell"
        >
          <span
            class="emotion-9"
          >
            VERIFIED
          </span>
        </td>
        <td
          class="emotion-10"
          role="gridcell"
        >
          <div
            class="chakra-stack emotion-11"
          >
            <button
              class="chakra-button emotion-12"
              data-testid="member-modal-button"
              type="button"
            >
              <svg
                fill="currentColor"
                font-size="24px"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M16 8s-3-5.5-8-5.5S0 8 0 8s3 5.5 8 5.5S16 8 16 8zM1.173 8a13.133 13.133 0 0 1 1.66-2.043C4.12 4.668 5.88 3.5 8 3.5c2.12 0 3.879 1.168 5.168 2.457A13.133 13.133 0 0 1 14.828 8c-.058.087-.122.183-.195.288-.335.48-.83 1.12-1.465 1.755C11.879 11.332 10.119 12.5 8 12.5c-2.12 0-3.879-1.168-5.168-2.457A13.134 13.134 0 0 1 1.172 8z"
                />
                <path
                  d="M8 5.5a2.5 2.5 0 1 0 0 5 2.5 2.5 0 0 0 0-5zM4.5 8a3.5 3.5 0 1 1 7 0 3.5 3.5 0 0 1-7 0z"
                />
              </svg>
            </button>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
